test/select.jl

```diff
@@ -0,0 +1,838 @@
+module TestSelect
+
+using DataFrames, Test, Random
+
+Random.seed!(1234)
+
+@testset "select! Not" begin
+    df = DataFrame(a=1, b=2, c=3, d=4, e=5)
+    @test_throws BoundsError select!(df, Not(0))
+    @test_throws BoundsError select!(df, Not(6))
+    @test_throws ArgumentError select!(df, Not([1, 1]))
+    @test_throws ArgumentError select!(df, Not(:f))
+    @test_throws BoundsError select!(df, Not([true, false]))
+
+    d = copy(df)
+    select!(d, Not([:a, :e, :c]))
+    @test names(d) == [:b, :d]
```

Comment:
Check values instead of just names while you're at it?

Reply:
done